openedx/core/djangoapps/schedules/management/commands/__init__.py

```diff
@@ -49,19 +66,23 @@ def handle(self, *args, **options):
             tzinfo=pytz.UTC
         )
         self.log_debug('Current date = %s', current_date.isoformat())
+        override_recipient_email = options.get('override_recipient_email')
+        override_middlewares = options.get('override_middlewares')
 
-        site = Site.objects.get(domain__iexact=options['site_domain_name'])
-        self.log_debug('Running for site %s', site.domain)
+        site_domain_name = options['site_domain_name']
+        site_filter = Q(domain__iexact=site_domain_name) if site_domain_name else Q()
 
-        override_recipient_email = options.get('override_recipient_email')
-        self.send_emails(site, current_date, override_recipient_email)
+        for site in Site.objects.filter(site_filter):
```

Comment:
A small nit: this fails silently if the provided domain name doesn't match any site. It would be handy for it to error if no sites matched.

Reply:
This is intended to be configured as a cron job, so I would expect this to be configured (and tested) once and run indefinitely. We can certainly add an extra check there, but that would mean this check runs every time the cron job runs, and I am not sure if that would be very useful.

However, let me know if you feel strongly about it.

Reply:
I don't feel too strongly. I was thinking just something along the lines of (pseudocode):

```suggestion
        sites = Site.objects.filter(site_filter)
        if len(sites) == 0:
            raise Error("site not found")
        for site in Site.objects.filter(site_filter):
```

Just so the user has some feedback if there's a typo in the domain for example.

Reply:
I'm not sure what the behaviour was on a site not found before this patch. But feel free to merge as-is; as you say, it's not high priority.

Reply:
@samuelallan72  Cool, I added an info log if no site is found.